bin/setup_new

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+# taken from https://gist.github.com/gvarela/9370838
+
+{
+  RUBY_VERSION=$(cat .ruby-version)
+  RUBY_VERSION=${RUBY_VERSION:=2.3.0}
+  PROJECT_NAME="ifme"
+} >/dev/null 2>&1
+
+## bootstrap function called at bottom of file comment out verification steps you don't need
+init (){
+  verify_ruby_version &&
+    verify_bundled_gems
+  verify_env_file
+  copy_example_config_files
+  verify_postgres_running &&
+    verify_database_migrations
+}
+
+
+## utility functions to test dependencies
+is_darwin (){
+  PLATFORM=$(uname)
+  [ "$PLATFORM" == "Darwin" ]
+}
+
+has_homebrew (){
+  [ is_darwin ] && [ hash brew >/dev/null 2>&1 ]
+}
+
+verify_homebrew (){
+  if [ ! has_homebrew ]
+  then
+    if [ is_darwin ]
+    then
+      echo 'Please install homebrew. Install instructions at http://brew.sh/'
+    else
+      echo "Please install Ruby version $RUBY_VERSION"
+    fi
+    exit 1
+  fi
+}
+
+has_postgres (){
+  hash psql >/dev/null 2>&1
+}
+
+postgres_is_running (){
+  ps aux | grep [p]ostgres >/dev/null 2>&1
+}
+
+has_rvm (){
+  hash rvm >/dev/null 2>&1
+}
+
+update_rvm (){
+  rvm get latest
+}
+
+rvm_install_ruby_version () {
+  rvm install $RUBY_VERSION
+}
+
+has_rbenv (){
+  hash rbenv >/dev/null 2>&1
+}
+
+update_rbenv (){
+  brew update
+  brew upgrade ruby-build
+}
+
+rbenv_install_ruby_version () {
+  rbenv install $RUBY_VERSION
+  rbenv local $RUBY_VERSION
+}
+
+has_ruby_version (){
+  INSTALLED_RUBY_VERSION=$(ruby -e 'puts RUBY_VERSION')
+  [ "$INSTALLED_RUBY_VERSION" == "$RUBY_VERSION" ]
+}
+
+install_ruby_version (){
+  verify_homebrew
+
+  if [ ! has_rvm ] && [ ! has_rbenv ]
+  then
+    echo 'Please install rbenv or rvm to manage your ruby versions.'
+    exit 1
+  fi
+
+  echo "Installing Ruby Version $RUBY_VERSION"
+  has_rvm && {
+  update_rvm
+  rvm_install_ruby_version
+}
+
+has_rbenv && {
+update_rbenv
+rbenv_install_ruby_version
+  }
+}
+
+verify_ruby_version (){
+  has_ruby_version || install_ruby_version
+}
+
+verify_bundled_gems () {
+  echo 'Updating gems...'
+  {
+    (gem list -i bundler || gem install bundler) && bundle install
+  } 2>&1
+}
+
+verify_env_file (){
+  if [ ! -f .env ]
+  then
+    echo "Generating secret..."
+    SECRET=$(bundle exec rake secret)
+    echo "SECRET_KEY_BASE=$SECRET" >> .env
```

Comment:
Good catch. I'm so used to using dotenv that I didn't realize that we actually needed the dotenv gem for that to work.

Appending to the config file makes me worry that the secret will be appended more than once if for some reason if the setup script is run more than once (which would probably only happen if the script were unsuccessful and needed to be rerun). I guess appending `SECRET_KEY_BASE: "#{generated secret key}"` more than once wouldn't break anything, I assume the last one would just override any previous declarations.

[This post](http://daniel.fone.net.nz/blog/2013/05/20/a-better-way-to-manage-the-rails-secret-token/) suggests hardcoding the secret key for dev and test environments, since it's really only production that needs a secure key.

```
# config/initializers/secret_token.rb

MyApp::Application.config.secret_token = if Rails.env.development? or Rails.env.test?
  ('x' * 30) # meets minimum requirement of 30 chars long
else
  ENV['SECRET_TOKEN']
end
```

Or since Figaro allows environment-specific declarations we could add hard-coded dev and test secrets to `config/application.example.yml`

```
development:
  SECRET_KEY_BASE: "#{something_that_is_30_characters}"
```